### ISSUE TITLE
race condition between sr_audit in background and stop, causes processes to be restarted while stopping

### ISSUE THREAD

**Issue by benlapETS** [MetPX/sarracenia#210]
The cmd to stop shovel dd2 was clearly triggered in flow_limit:
```
Initial sample building sample size 351 need at least 3000 
Sample now:   3099 Missed_dispositions:0
Sufficient!
Stopping shovels and waiting...
2019-05-29 12:03:58,397 [INFO] sr_shovel t_dd2_f00 01 stopped
2019-05-29 12:03:58,398 [INFO] sr_shovel t_dd2_f00 02 stopped
2019-05-29 12:03:58,426 [INFO] sr_shovel t_dd2_f00 03 stopped
2019-05-29 12:03:58,426 [INFO] sr_shovel t_dd2_f00 04 stopped
2019-05-29 12:03:58,426 [INFO] sr_shovel t_dd2_f00 05 stopped
2019-05-29 12:04:00,790 [INFO] sr_shovel t_dd1_f00 01 stopped
2019-05-29 12:04:00,846 [INFO] sr_shovel t_dd1_f00 02 stopped
2019-05-29 12:04:00,992 [INFO] sr_shovel t_dd1_f00 03 stopped
2019-05-29 12:04:01,001 [INFO] sr_shovel t_dd1_f00 04 stopped
2019-05-29 12:04:01,005 [INFO] sr_shovel t_dd1_f00 05 stopped
```
But it seems that dd1 stopped but dd2 continued shovelling right after because of 2 rebel instances:
```
sr_shovel_t_dd2_f00_02.log 2019-05-29 13:18:11,574 [INFO] msg_total: 14785 messages received:   3.1 msg/s, 221.0K bytes/s, lag:   17 s
sr_shovel_t_dd2_f00_03.log 2019-05-29 13:18:18,996 [INFO] msg_total: 16126 messages received:   3.4 msg/s, 251.2K bytes/s, lag:   18 s
...
test  1 FAILURE: sr_shovel	 (3145) t_dd1 should have the same number of items as t_dd2	 (31190)
```
We can also understand that directly from those 2 instances logs:
```
sr_shovel_t_dd2_f00_02.log:2019-05-29 12:03:56,375 [INFO] signal stop (SIGTERM)
sr_shovel_t_dd2_f00_02.log:2019-05-29 12:03:56,375 [INFO] sr_shovel stop
sr_shovel_t_dd2_f00_02.log:2019-05-29 12:03:57,572 [INFO] sr_shovel t_dd2_f00 start
```
```
sr_shovel_t_dd2_f00_03.log:2019-05-29 12:03:56,375 [INFO] signal stop (SIGTERM)
sr_shovel_t_dd2_f00_03.log:2019-05-29 12:03:56,375 [INFO] sr_shovel stop
sr_shovel_t_dd2_f00_03.log:2019-05-29 12:03:57,572 [INFO] sr_shovel t_dd2_f00 start
```
Also from the stdout of the instance itself, we see part of the explanation (it flags both instances as missing and restart it):
```
flowsetup_f00.log:2019-05-29 12:03:57,382 [INFO] sr_shovel t_dd2_f00 02 instance missing (pid=29989)
flowsetup_f00.log:2019-05-29 12:03:57,382 [INFO] sr_shovel t_dd2_f00 02 restart
flowsetup_f00.log:2019-05-29 12:03:57,382 [INFO] sr_shovel t_dd2_f00 02 stopping (pid=29989)
flowsetup_f00.log:2019-05-29 12:03:57,382 [INFO] sr_shovel t_dd2_f00 02 stopped
flowsetup_f00.log:2019-05-29 12:03:57,382 [INFO] sr_shovel t_dd2_f00 02 starting
flowsetup_f00.log:2019-05-29 12:03:57,384 [INFO] sr_shovel t_dd2_f00 03 instance missing (pid=29990)
flowsetup_f00.log:2019-05-29 12:03:57,384 [INFO] sr_shovel t_dd2_f00 03 restart
flowsetup_f00.log:2019-05-29 12:03:57,384 [INFO] sr_shovel t_dd2_f00 03 stopping (pid=29990)
flowsetup_f00.log:2019-05-29 12:03:57,384 [INFO] sr_shovel t_dd2_f00 03 stopped
flowsetup_f00.log:2019-05-29 12:03:57,384 [INFO] sr_shovel t_dd2_f00 03 starting
```

**Comment by petersilva** [comment]
now that the WIP for #174 has been merged, the time to run sr stop has been reduced drastically.  We should check this window should be substantially reduced.  on the other hand... Int that refactored sr.py, the sr_audit process is killed after the instance pids.  If it were killed before the instance pids, that would eliminate this issue entirely.

**Comment by petersilva** [comment]
likely fixed by: 7a39fe217240758e3481375f9960581ba2be2382

**Comment by petersilva** [comment]
fix released 2.19.09
